OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -129,6 +133,7 @@ public void Tick(ActorMap am)
 				var delta = new WVec(range, range, WDist.Zero);
 				currentActors = am.ActorsInBox(position - delta, position + delta)
 					.Where(a => (a.CenterPosition - position).HorizontalLengthSquared < range.LengthSquared)
+					.Where(a => vRange.Length == 0 || (a.World.Map.DistanceAboveTerrain(a.CenterPosition).LengthSquared <= vRange.LengthSquared))
```

Comment:
I'm not sure how smart Linq is about these expressions, but just to be safe I think it would be better to use only a single `.Where` and combine both conditions, lest the list is walked twice.